checker/src/org/checkerframework/checker/lowerbound/LowerBoundAnnotatedTypeFactory.java

```diff
@@ -0,0 +1,653 @@
+         */
+        public void promoteType(AnnotatedTypeMirror typeSrc, AnnotatedTypeMirror typeDst) {
+            if (typeSrc.hasAnnotation(POS)) {
+                typeDst.replaceAnnotation(POS);
+            } else if (typeSrc.hasAnnotation(NN)) {
+                typeDst.replaceAnnotation(POS);
+            } else if (typeSrc.hasAnnotation(GTEN1)) {
+                typeDst.replaceAnnotation(NN);
+            } else { //Only unknown is left.
+                typeDst.replaceAnnotation(UNKNOWN);
+            }
+            return;
+        }
+
+        /**
+         *  Sets typeDst to the immediate subtype of typeSrc, unless typeSrc is already
+         *  LowerBoundUnknown. Implements the following transitions:
+         *  <pre>
+         *       pos &rarr; nn
+         *       nn &rarr; gte-1
+         *       gte-1, lbu &rarr; lbu
+         *  </pre>
+         */
+        public void demoteType(AnnotatedTypeMirror typeSrc, AnnotatedTypeMirror typeDst) {
+            if (typeSrc.hasAnnotation(POS)) {
+                typeDst.replaceAnnotation(NN);
+            } else if (typeSrc.hasAnnotation(NN)) {
+                typeDst.replaceAnnotation(GTEN1);
+            } else { // GTEN1 and UNKNOWN both become UNKNOWN.
+                typeDst.replaceAnnotation(UNKNOWN);
+            }
+            return;
+        }
+
+        /**
+         *  Determine the annotation that should be associated with a literal.
+         */
+        private AnnotationMirror anmFromVal(int val) {
+            if (val == -1) {
+                return GTEN1;
+            } else if (val == 0) {
+                return NN;
+            } else if (val > 0) {
+                return POS;
+            } else {
+                return UNKNOWN;
+            }
+        }
+
+        @Override
+        public Void visitLiteral(LiteralTree tree, AnnotatedTypeMirror type) {
+            // We could call the constant Value Checker here but if we already know it's a literal...
+            if (tree.getKind() == Tree.Kind.INT_LITERAL
+                    || tree.getKind() == Tree.Kind.LONG_LITERAL
+                    || tree.getKind() == Tree.Kind.CHAR_LITERAL) {
+                int val = (int) tree.getValue();
+                type.addAnnotation(anmFromVal(val));
+            }
+            return super.visitLiteral(tree, type);
+        }
+
+        /**
+         *  Call increment and decrement helper functions.
+         */
+        @Override
+        public Void visitUnary(UnaryTree tree, AnnotatedTypeMirror typeDst) {
+            AnnotatedTypeMirror typeSrc = getAnnotatedType(tree.getExpression());
+            switch (tree.getKind()) {
+                case PREFIX_INCREMENT:
+                    promoteType(typeSrc, typeDst);
+                    break;
+                case PREFIX_DECREMENT:
+                    demoteType(typeSrc, typeDst);
+                    break;
+                case POSTFIX_INCREMENT: // Do nothing. The hack above takes care of these.
```

Comment:
Once you remove the hack, you'll still want to do nothing, but you should change the comment.